src/TaskForm.tsx

```diff
@@ -0,0 +1,39 @@
+import { ContextType, useContext } from "react";
+import { TodoContext } from './context/tasks'
+
+const TaskForm = () => {
+  const todos = useContext(TodoContext)   ; 
+
+  console.log(todos)
+ 
+  return (
+    <div>
+      {/* <div> selected task : {selectedTask} </div> */}
+      <div>
+        taskslist:
+        {/* <ul>
+          {todos.map((toe: any) => (
+            <li> {toe} </li>
+          ))}
+        </ul> */}
+      </div>
+      <div>
+        <input id = "taskinfo" />
+        <button onClick = {()=> {
+            let taskvalue = (document.getElementById("taskinfo") as unknown as HTMLInputElement).value;
+            
+            console.log(taskvalue)
+            if (taskvalue ) {
+
+              // let newresult = [...tasks ,taskvalue]
+              // setTasks(newresult)
+            }
+            
+        }}>save </button>
+        
```

Comment:
Have a separate method for any event calls and call that function in events
Use onBlur or onChange method to save value in state and use that don't capture value of input like this
And Whats the use if statement here if its not required or doing anything then remove these things.
Remove any consoles, commented and unused code.